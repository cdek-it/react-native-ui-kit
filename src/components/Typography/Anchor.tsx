import type { Icon } from '@tabler/icons-react-native'
import { Fragment, memo, useCallback, useMemo, useState } from 'react'
import {
  Pressable,
  type StyleProp,
  Text,
  type TextProps,
  type TextStyle,
  View,
} from 'react-native'

import { makeStyles } from '../../utils/makeStyles'

const WORD_JOINER = '\u2060' // символ невидимого пробела, чтобы избежать разрыва строки между текстом и иконкой

export interface AnchorProps
  extends Omit<TextProps, 'onPressIn' | 'onPressOut'> {
  readonly onPress: () => void
  /** true, если необходим базовый размер текста ссылки, равный 14 */
  readonly base?: boolean
  /** true, если необходимо состояние посещенной ссылки. Меняет цвет */
  readonly visited?: boolean
  /**
   * Используется для рендера ссылки в составе другого текста.
   * Если true, компонент оборачивается во фрагмент вместо обычного View
   */
  readonly noWrapper?: boolean
  /** Кастомные стили, применяемые к тексту */
  readonly style?: StyleProp<TextStyle>
  readonly LeftIcon?: Icon
  readonly RightIcon?: Icon
}

/**
 * Используется для ссылок и якорей
 * @see https://www.figma.com/design/2ZnL6XPKEpxAHvrlbRvnMu/Template-Tailwind-CSS--DS-?node-id=1-271
 */
export const Anchor = memo(
  ({
    onPress,
    base,
    visited,
    children,
    noWrapper,
    LeftIcon,
    RightIcon,
    testID,
    style,
    ...other
  }: AnchorProps) => {
    const styles = useStyles()

    const [pressed, setPressed] = useState(false)
    const onPressIn = useCallback(() => setPressed(true), [])
    const onPressOut = useCallback(() => setPressed(false), [])

    const Wrapper = noWrapper ? Fragment : View
    const containerProps = useMemo(() => {
      if (noWrapper) {
        return {}
      }

      return { style: styles.container, testID: testID || AnchorTestId.root }
    }, [noWrapper, styles.container, testID])

    const iconColor = useMemo(
      () => (visited ? styles.visited.color : styles.text.color),
      [styles.text.color, styles.visited.color, visited]
    )

    return (
      <Wrapper {...containerProps}>
        {LeftIcon ? (
          <Pressable
            style={styles.leftIconContainer}
            testID={AnchorTestId.leftPressable}
            onPress={onPress}
            onPressIn={onPressIn}
            onPressOut={onPressOut}
          >
            <LeftIcon
              testID={AnchorTestId.leftIcon}
              {...styles.icon}
              color={iconColor}
            />
          </Pressable>
        ) : null}
        <Text
          suppressHighlighting
          style={[
            styles.text,
            pressed && styles.underlined,
            base && styles.base,
            visited && styles.visited,
            style,
          ]}
          testID={AnchorTestId.text}
          onPress={onPress}
          onPressIn={onPressIn}
          onPressOut={onPressOut}
          {...other}
        >
          {LeftIcon ? WORD_JOINER : null}
          {children}
          {RightIcon ? WORD_JOINER : null}
        </Text>

        {RightIcon ? (
          <Pressable
            style={styles.rightIconContainer}
            testID={AnchorTestId.rightPressable}
            onPress={onPress}
            onPressIn={onPressIn}
            onPressOut={onPressOut}
          >
            <RightIcon
              {...styles.icon}
              color={iconColor}
              testID={AnchorTestId.rightIcon}
            />
          </Pressable>
        ) : null}
      </Wrapper>
    )
  }
)

const useStyles = makeStyles(({ spacing, typography }) => ({
  container: { flexDirection: 'row', alignItems: 'center' },
  text: {
    flexShrink: 1,
    fontSize: 12.25,
    includeFontPadding: false,
    verticalAlign: 'middle',
    color: typography.Color.Service['text-info'],
  },
  underlined: { textDecorationLine: 'underline' },
  base: { fontSize: 14 },
  visited: { color: typography.Color.Service['text-help'] },
  icon: { width: 14, height: 14 },
  leftIconContainer: { paddingRight: spacing.Padding['p-2'] },
  rightIconContainer: { paddingLeft: spacing.Padding['p-2'] },
}))

export const AnchorTestId = {
  root: 'Anchor',
  leftPressable: 'AnchorLeftPressable',
  rightPressable: 'AnchorRightPressable',
  leftIcon: 'LeftIcon',
  rightIcon: 'RightIcon',
  text: 'AnchorText',
}
