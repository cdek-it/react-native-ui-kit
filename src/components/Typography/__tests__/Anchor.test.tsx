import { IconLink } from '@tabler/icons-react-native'
import { fireEvent, render, userEvent } from '@testing-library/react-native'

import { Anchor, type AnchorProps, AnchorTestId } from '../Anchor'

describe('Body component tests', () => {
  const defaultProps: AnchorProps = { onPress: jest.fn() }

  test('default props', () => {
    const renderedAnchor = render(<Anchor {...defaultProps}>link</Anchor>)

    expect(renderedAnchor.toJSON()).toMatchSnapshot()
  })

  test('левая иконка, base, visited, testID = CustomAnchorId', () => {
    const renderedAnchor = render(
      <Anchor
        {...defaultProps}
        base
        visited
        LeftIcon={IconLink}
        testID='CustomAnchorId'
      >
        link
      </Anchor>
    )

    expect(renderedAnchor.toJSON()).toMatchSnapshot()
  })

  test('правая иконка, noWrapper', () => {
    const renderedAnchor = render(
      <Anchor {...defaultProps} noWrapper RightIcon={IconLink}>
        link
      </Anchor>
    )

    expect(renderedAnchor.toJSON()).toMatchSnapshot()
  })

  test('левая иконка, правая иконка', () => {
    const renderedAnchor = render(
      <Anchor {...defaultProps} LeftIcon={IconLink} RightIcon={IconLink}>
        link
      </Anchor>
    )

    expect(renderedAnchor.toJSON()).toMatchSnapshot()
  })

  test('при нажатии на левую иконку текст подчеркивается', () => {
    const { getByTestId } = render(
      <Anchor {...defaultProps} LeftIcon={IconLink}>
        link
      </Anchor>
    )

    const leftIconPressable = getByTestId(AnchorTestId.leftPressable)
    fireEvent(leftIconPressable, 'pressIn')

    expect(getByTestId(AnchorTestId.text)).toHaveStyle({
      textDecorationLine: 'underline',
    })

    fireEvent(leftIconPressable, 'pressOut')

    expect(getByTestId(AnchorTestId.text)).not.toHaveStyle({
      textDecorationLine: 'underline',
    })
  })

  test('при нажатии на правую иконку текст подчеркивается', () => {
    const { getByTestId } = render(
      <Anchor {...defaultProps} RightIcon={IconLink}>
        link
      </Anchor>
    )

    const rightIconPressable = getByTestId(AnchorTestId.rightPressable)
    fireEvent(rightIconPressable, 'pressIn')

    expect(getByTestId(AnchorTestId.text)).toHaveStyle({
      textDecorationLine: 'underline',
    })

    fireEvent(rightIconPressable, 'pressOut')

    expect(getByTestId(AnchorTestId.text)).not.toHaveStyle({
      textDecorationLine: 'underline',
    })
  })

  test('при нажатии на текст текст подчеркивается', () => {
    const { getByTestId } = render(<Anchor {...defaultProps}>link</Anchor>)

    const text = getByTestId(AnchorTestId.text)
    fireEvent(text, 'pressIn')

    expect(getByTestId(AnchorTestId.text)).toHaveStyle({
      textDecorationLine: 'underline',
    })

    fireEvent(text, 'pressOut')

    expect(getByTestId(AnchorTestId.text)).not.toHaveStyle({
      textDecorationLine: 'underline',
    })
  })

  test('при нажатии на левую, правую иконки, либо на текст, вызывается onPress', async () => {
    const mockOnPress = jest.fn()
    const { getByTestId } = render(
      <Anchor LeftIcon={IconLink} RightIcon={IconLink} onPress={mockOnPress}>
        link
      </Anchor>
    )
    const user = userEvent.setup()

    expect(mockOnPress).not.toHaveBeenCalled()

    await user.press(getByTestId(AnchorTestId.leftPressable))
    await user.press(getByTestId(AnchorTestId.rightPressable))
    await user.press(getByTestId(AnchorTestId.text))

    expect(mockOnPress).toHaveBeenCalledTimes(3)
  })
})
