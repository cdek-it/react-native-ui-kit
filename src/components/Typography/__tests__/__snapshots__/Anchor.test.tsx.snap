// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body component tests default props 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
  testID="Anchor"
>
  <Text
    onPress={[MockFunction]}
    onPressIn={[Function]}
    onPressOut={[Function]}
    style={
      [
        {
          "color": "#49a2ef",
          "flexShrink": 1,
          "fontSize": 12.25,
          "includeFontPadding": false,
          "verticalAlign": "middle",
        },
        false,
        undefined,
        undefined,
        undefined,
      ]
    }
    suppressHighlighting={true}
    testID="AnchorText"
  >
    link
  </Text>
</View>
`;

exports[`Body component tests левая иконка, base, visited, testID = CustomAnchorId 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
  testID="CustomAnchorId"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "paddingRight": 7,
      }
    }
    testID="AnchorLeftPressable"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={14}
      bbWidth={14}
      fill="none"
      focusable={false}
      height={14}
      meetOrSlice={0}
      minX={0}
      minY={0}
      stroke="#ab77df"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 14,
            "width": 14,
          },
        ]
      }
      testID="LeftIcon"
      vbHeight={24}
      vbWidth={24}
      width={14}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4289427423,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2}
      >
        <RNSVGPath
          d="M9 15l6 -6"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4289427423,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="LeftIcon"
        />
        <RNSVGPath
          d="M11 6l.463 -.536a5 5 0 0 1 7.071 7.072l-.534 .464"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4289427423,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="LeftIcon"
        />
        <RNSVGPath
          d="M13 18l-.397 .534a5.068 5.068 0 0 1 -7.127 0a4.972 4.972 0 0 1 0 -7.071l.524 -.463"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4289427423,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="LeftIcon"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    onPress={[MockFunction]}
    onPressIn={[Function]}
    onPressOut={[Function]}
    style={
      [
        {
          "color": "#49a2ef",
          "flexShrink": 1,
          "fontSize": 12.25,
          "includeFontPadding": false,
          "verticalAlign": "middle",
        },
        false,
        {
          "fontSize": 14,
        },
        {
          "color": "#ab77df",
        },
        undefined,
      ]
    }
    suppressHighlighting={true}
    testID="AnchorText"
  >
    ⁠
    link
  </Text>
</View>
`;

exports[`Body component tests левая иконка, правая иконка 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
  testID="Anchor"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "paddingRight": 7,
      }
    }
    testID="AnchorLeftPressable"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={14}
      bbWidth={14}
      fill="none"
      focusable={false}
      height={14}
      meetOrSlice={0}
      minX={0}
      minY={0}
      stroke="#49a2ef"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 14,
            "width": 14,
          },
        ]
      }
      testID="LeftIcon"
      vbHeight={24}
      vbWidth={24}
      width={14}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4283015919,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2}
      >
        <RNSVGPath
          d="M9 15l6 -6"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4283015919,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="LeftIcon"
        />
        <RNSVGPath
          d="M11 6l.463 -.536a5 5 0 0 1 7.071 7.072l-.534 .464"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4283015919,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="LeftIcon"
        />
        <RNSVGPath
          d="M13 18l-.397 .534a5.068 5.068 0 0 1 -7.127 0a4.972 4.972 0 0 1 0 -7.071l.524 -.463"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4283015919,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="LeftIcon"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    onPress={[MockFunction]}
    onPressIn={[Function]}
    onPressOut={[Function]}
    style={
      [
        {
          "color": "#49a2ef",
          "flexShrink": 1,
          "fontSize": 12.25,
          "includeFontPadding": false,
          "verticalAlign": "middle",
        },
        false,
        undefined,
        undefined,
        undefined,
      ]
    }
    suppressHighlighting={true}
    testID="AnchorText"
  >
    ⁠
    link
    ⁠
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "paddingLeft": 7,
      }
    }
    testID="AnchorRightPressable"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={14}
      bbWidth={14}
      fill="none"
      focusable={false}
      height={14}
      meetOrSlice={0}
      minX={0}
      minY={0}
      stroke="#49a2ef"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 14,
            "width": 14,
          },
        ]
      }
      testID="RightIcon"
      vbHeight={24}
      vbWidth={24}
      width={14}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4283015919,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2}
      >
        <RNSVGPath
          d="M9 15l6 -6"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4283015919,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="RightIcon"
        />
        <RNSVGPath
          d="M11 6l.463 -.536a5 5 0 0 1 7.071 7.072l-.534 .464"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4283015919,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="RightIcon"
        />
        <RNSVGPath
          d="M13 18l-.397 .534a5.068 5.068 0 0 1 -7.127 0a4.972 4.972 0 0 1 0 -7.071l.524 -.463"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4283015919,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="RightIcon"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`Body component tests правая иконка, noWrapper 1`] = `
[
  <Text
    onPress={[MockFunction]}
    onPressIn={[Function]}
    onPressOut={[Function]}
    style={
      [
        {
          "color": "#49a2ef",
          "flexShrink": 1,
          "fontSize": 12.25,
          "includeFontPadding": false,
          "verticalAlign": "middle",
        },
        false,
        undefined,
        undefined,
        undefined,
      ]
    }
    suppressHighlighting={true}
    testID="AnchorText"
  >
    link
    ⁠
  </Text>,
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "paddingLeft": 7,
      }
    }
    testID="AnchorRightPressable"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={14}
      bbWidth={14}
      fill="none"
      focusable={false}
      height={14}
      meetOrSlice={0}
      minX={0}
      minY={0}
      stroke="#49a2ef"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 14,
            "width": 14,
          },
        ]
      }
      testID="RightIcon"
      vbHeight={24}
      vbWidth={24}
      width={14}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4283015919,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2}
      >
        <RNSVGPath
          d="M9 15l6 -6"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4283015919,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="RightIcon"
        />
        <RNSVGPath
          d="M11 6l.463 -.536a5 5 0 0 1 7.071 7.072l-.534 .464"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4283015919,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="RightIcon"
        />
        <RNSVGPath
          d="M13 18l-.397 .534a5.068 5.068 0 0 1 -7.127 0a4.972 4.972 0 0 1 0 -7.071l.524 -.463"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4283015919,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
          testID="RightIcon"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>,
]
`;
